q.ml

```diff
@@ -298,26 +298,78 @@ let to_float x =
          correctly-rounded result. *)
       Int64.to_float (Z.to_int64 p) /. Int64.to_float (Z.to_int64 q)
     else begin
-      (* |p| is in [2^(np-1), 2^np)
+      let negat =
+        if Z.lt p Z.zero then -1 else 1
```

Comment:
Using `Z.sign` is slightly faster, but it probably makes no measurable difference.
```suggestion
        if Z.sign p < 0 then -1 else 1
```